src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -239,34 +257,63 @@ bool emberAfGeneralCommissioningClusterCommissioningCompleteCallback(
         }
         else
         {
-            if (failSafe.NocCommandHasBeenInvoked())
+            CHIP_ERROR err;
+
+#if CHIP_CONFIG_TC_REQUIRED
+
+            uint16_t termsAndConditionsAcceptedAcknowledgements;
+            bool hasRequiredTermAccepted;
+            bool hasRequiredTermVersionAccepted;
+
+            err = enhancedSetupFlowProvider->GetTermsAndConditionsAcceptedAcknowledgements(
+                termsAndConditionsAcceptedAcknowledgements);
+            CheckSuccess(err, Failure);
+
+            err = enhancedSetupFlowProvider->HasTermsAndConditionsRequiredAcknowledgementsBeenAccepted(hasRequiredTermAccepted);
+            CheckSuccess(err, Failure);
+
+            err = enhancedSetupFlowProvider->HasTermsAndConditionsRequiredAcknowledgementsVersionBeenAccepted(
+                hasRequiredTermVersionAccepted);
+            CheckSuccess(err, Failure);
+
+            if (!hasRequiredTermAccepted)
             {
-                CHIP_ERROR err = fabricTable.CommitPendingFabricData();
-                if (err != CHIP_NO_ERROR)
-                {
-                    // No need to revert on error: CommitPendingFabricData always reverts if not fully successful.
-                    ChipLogError(FailSafe, "GeneralCommissioning: Failed to commit pending fabric data: %" CHIP_ERROR_FORMAT,
-                                 err.Format());
-                }
-                else
+                ChipLogProgress(AppServer, "Required terms and conditions have not been accepted");
+                Breadcrumb::Set(commandPath.mEndpointId, 0);
+                response.errorCode = (0 == termsAndConditionsAcceptedAcknowledgements)
+                    ? CommissioningErrorEnum::kTCAcknowledgementsNotReceived
+                    : CommissioningErrorEnum::kRequiredTCNotAccepted;
+            }
+
+            else if (!hasRequiredTermVersionAccepted)
+            {
+                ChipLogProgress(AppServer, "Minimum terms and conditions version has not been accepted");
+                Breadcrumb::Set(commandPath.mEndpointId, 0);
+                response.errorCode = CommissioningErrorEnum::kTCMinVersionNotMet;
+            }
+
+            else
+#endif
```

Comment:
- Please move this into a helper whose body is is conditional on CHIP_CONFIG_TC_REQUIRED, to reduce the conditional thinking burden over this big pane of conditional code.
- At the end of each `#endif`, indicate define: `#endif  // CHIP_FIG_TC_REQUIRED`